### PR TITLE
fix(sdk): Fix compiler bug for legacy outputs mlpipeline-ui-metadata and mlpipeline-metrics. Fixes #6311 

### LINKED ISSUE #6311: [testing] postsubmit test failure 2021-08-12
https://oss-prow.knative.dev/view/gs/oss-prow/logs/kubeflow-pipeline-postsubmit-standalone-component-test/1425587097510612992#1:build-log.txt%3A5394

For standalone components test, it fails when building sample test image:

> sample-test-rb7fv-2769807467: error: google-api-core 1.26.0 is installed but google-api-core<3.0dev,>=1.29.0 is required by {'google-cloud-storage'}
sample-test-rb7fv-2769807467: The command '/bin/sh -c cd /sdk/python && python3 setup.py install' returned a non-zero code: 1

### PR DESCRIPTION
**Pull request by neuromage**
Fixes the bug introduced by PR #6268 

Fixes #6311

### CODE REVIEW

File: sdk/python/kfp/dsl/_container_op.py [hunk 1]
```diff
@@ -1062,6 +1062,15 @@ def __init__(self, argument, input=None, path=None):
     self.path = path
 
 
+def _is_legacy_output_name(output_name: str) -> Tuple[bool, str]:
+    normalized_output_name = re.sub('[^a-zA-Z0-9]', '-', output_name.lower())
```
Comment (Bobgy) [inline sdk/python/kfp/dsl/_container_op.py:1066]: I think we have this transform in many places, shall we extract a common util?

**Review by Bobgy — COMMENTED** [review]
/lgtm

**Comment by Bobgy** [comment]
Thank you for fixing this quickly!

File: sdk/python/kfp/dsl/_container_op.py (hunk 1 above)
Comment (neuromage) [reply sdk/python/kfp/dsl/_container_op.py:1066]: I think that's what I just did here? :-)

Comment (Bobgy) [reply sdk/python/kfp/dsl/_container_op.py:1066]: what's the difference between this and https://github.com/kubeflow/pipelines/blob/e7c281bd65ca79c20c2ae52a4dc4798241333b53/sdk/python/kfp/compiler/_k8s_helper.py#L20?

Comment (Bobgy) [reply sdk/python/kfp/dsl/_container_op.py:1066]: and
* https://github.com/kubeflow/pipelines/blob/cc83e1089b573256e781ed2e4ac90f604129e769/sdk/python/kfp/components/_naming.py#L51-L53
* https://github.com/kubeflow/pipelines/blob/cc83e1089b573256e781ed2e4ac90f604129e769/sdk/python/kfp/dsl/_pipeline_param.py#L25-L41
* https://github.com/kubeflow/pipelines/blob/fdab2e679cad415871b8d864902e1b7e1e292451/sdk/python/kfp/dsl/dsl_utils.py#L51-L58

looks like there's an implementation everywhere...

this is a non blocker for this PR

**Comment by Bobgy** [comment]
/approve
let's get the fix in first

File: sdk/python/kfp/dsl/_container_op.py (hunk 1 above)
Comment (neuromage) [reply sdk/python/kfp/dsl/_container_op.py:1066]: Sorry, I don't see your point here. These different functions are used in different places. I don't think the goal of this PR is to refactor and clean up this code.

**Pull request merged**